enterprise_access/apps/api/v1/views.py

```diff
@@ -134,6 +159,119 @@ def _validate_subsidy_request(self):
+
+        subsidy_request_uuid should be a valid uuid.
+        """
+        if not subscription_plan_uuid:
+            error_msg = (
+                'You must provide a License Subscription UUID so subsidy requests '
+                'can be completed (so a License can be assigned to the user).'
+            )
+            logger.exception(error_msg)
+            raise SubsidyRequestError(error_msg, status.HTTP_422_UNPROCESSABLE_ENTITY)
+
+        try:
+            validate_uuid(subscription_plan_uuid)
+        except ParseError as exc:
+            error_msg = f'Subscription Plan UUID provided ({subscription_plan_uuid}) is not a valid UUID'
+            logger.exception(error_msg)
+            raise SubsidyRequestError(error_msg, status.HTTP_422_UNPROCESSABLE_ENTITY) from exc
+
+    def _verify_subsidies_remaining(self, subscription_plan_uuid, subsidy_request_uuids):
+        """
+        Check license manager to make sure there are subscriptions remaining.
+        """
+        client = LicenseManagerApiClient()
+        subscription_overview = client.get_subscription_overview(subscription_plan_uuid)
+        status_counts = subscription_overview.get('results')
+
+        remaining_unassigned_codes = 0
+        for status_count in status_counts:
+            if status_count['status'] == 'unassigned':
+                remaining_unassigned_codes = status_count['count']
+                break
+
+        if remaining_unassigned_codes < len(subsidy_request_uuids):
+            error_msg = (
+                f'Not enough licenses available for subscription {subscription_plan_uuid} '
+                'to approve the requests'
+            )
+            raise SubsidyRequestError(error_msg, status.HTTP_422_UNPROCESSABLE_ENTITY)
+
+    @permission_required(
+        constants.REQUESTS_ADMIN_LEARNER_ACCESS_PERMISSION,
+        fn=get_enterprise_uuid_from_request_data,
+    )
+    @action(detail=False, url_path='approve', methods=['post'])
+    def approve(self, *args, **kwargs):
+        """ Action of approving a License Subsidy Request """
+
+        subsidy_request_uuids = self.request.data.get('subsidy_request_uuids')
+        subscription_plan_uuid = self.request.data.get('subscription_plan_uuid')
+        reviewer_lms_user_id = self.lms_user_id
+
+        try:
+            self._validate_subsidy_request_uuids(subsidy_request_uuids)
+            self._validate_subscription_plan_uuid(subscription_plan_uuid)
+            self._verify_subsidies_remaining(subscription_plan_uuid, subsidy_request_uuids)
+        except SubsidyRequestError as exc:
+            logger.exception(exc)
+            return Response(exc.message, exc.http_status_code)
+
+        subsidy_requests = LicenseRequest.objects.filter(uuid__in=subsidy_request_uuids)
+        with transaction.atomic():
+            for subsidy_request in subsidy_requests:
+                if subsidy_request.state == SubsidyRequestStates.DENIED:
+                    error_msg = (
```

Comment:
[clarification]: if the subsidy request has already been approved, will we just ignore them in the task?

Reply:
we can, though it's probably better if we don't spin off a task at all. i can update this to keep track of which approval requests have already been approved so that we have the right set of things to run celery tasks for

Reply:
different approach: on deny/approve endpoints, we filter the uuids handed to us, and only actually act on things that are appropriate to act on

so if you had 5 UUIDs for approval, but 1 is already approved, we just approve 4 and do not error. and same for denying, if you go to deny a thing that is already denied, we simply do nothing instead of error

thoughts?

Reply:
hrm.... I wonder what the industry at large thinks about stuff like this? IMO to be semantically correct it should either be all or nothing. That said I'm not sure that provides the best experience either. It does seem OK to let the API return a 200 in the case when you have n requests but 1+ of those are already in the state you are attempting to set. That seems reasonable. 

What about partial applies? IF only one state is able to be changed, but the others not, what should we do there? Return a partial result? A partial error? for simplicity up front it would likely be better for us to make the failure case all or nothing. But I think we are already set for this as an exception raised would cause the atomic block to rollback.

I think that's my long winded way of saying I think this is ok....

Reply:
cool. i did partially mistype:

an admin tries to deny 4 requests, 1 of which has already been approved. error? or simply ignore the 1 approved and  deny the 3-not-yet-denied?

Reply:
> an admin tries to deny 4 requests, 1 of which has already been approved. error?

Yes I do think this is an error case. That said I don't believe/am hopeful the UI wouldn't allow this state to be submitted.